sklearn/inspection/_partial_dependence.py

```diff
@@ -148,22 +152,39 @@ def _partial_dependence_brute(est, grid, features, X, response_method):
                 X_eval[:, variable] = new_values[i]
 
         try:
-            predictions = prediction_method(X_eval)
+            # Note: predictions is of shape
+            # (n_points,) for non-multioutput regressors
+            # (n_points, n_tasks) for multioutput regressors
+            # (n_points, 1) for the regressors in cross_decomposition (I think)
+            # (n_points, 2) for binary classification
+            # (n_points, n_classes) for multiclass classification
+            pred = prediction_method(X_eval)
+
+            predictions.append(pred)
```

Comment:
Is it possible that this becomes prohibitively memory-hungry? How do we stop that?

Reply:
I feel it will not. The total size of `predictions` is, (`grid_resolution`, `n_points`, `x`). The default value for `grid_resolution` is a very sensible 100. `n_points` is the size of the dataset.  `x` is 1, 2, n_classes or n_tasks.